src/ploomber/util/util.py

```diff
@@ -82,6 +82,17 @@ def _make_requires_error_message(missing_pkgs, fn_name, extra_msg):
     return error_msg
 
 
+def check_mixed_envs():
+    try:
+        from pip._internal.operations import freeze
+    except ImportError:
+        from pip.operations import freeze
+    reqs = freeze.freeze()
+    for dep in reqs:
+        if ' @ file://' in dep:
+            raise CondaPipMixedEnvError(dep)
```

Comment:
maybe grab all the offending lines first so we show them all of them in the error message?